pypdf/filters.py

```diff
@@ -651,7 +647,7 @@ def decode(
         return tiff_header + data
 
 
-def decode_stream_data(stream: Any) -> Union[bytes, str]:  # utils.StreamObject
+def decode_stream_data(stream: Any) -> bytes:  # utils.StreamObject
```

Comment:
I do not think that we are going to change this, but `decode_stream_data` is a bit misleading here - I initially thought that this would return strings, given the name.

Reply:
For mypy analysis this change should be kept as now we are producting only bytes whereas previously it was a mix up

Reply:
Also this may induce wrongly users as an str will never be returned (any more)